fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java

```diff
@@ -294,11 +295,44 @@ private Void computeOlapScanNode(Operator node, ExpressionContext context, Table
         }
 
         builder.setOutputRowCount(tableRowCount);
+        if (isRewrittenMvGE(node, table, context)) {
+            adjustNestedMvStatistics(context.getGroupExpression().getGroup(), (MaterializedView) olapTable, builder);
+            if (node.getProjection() != null) {
+                builder.setShadowColumns(node.getProjection().getOutputColumns());
+            }
+        }
         // 4. estimate cardinality
         context.setStatistics(builder.build());
+
         return visitOperator(node, context);
     }
 
+    private void adjustNestedMvStatistics(Group group, MaterializedView mv, Statistics.Builder builder) {
+        for (Map.Entry<Long, List<Long>> entry : group.getRelatedMvs().entrySet()) {
+            if (entry.getValue().contains(mv.getId())) {
+                // mv is a nested mv based on entry.getKey
+                Optional<Statistics> baseStatistics = group.getMvStatistics(entry.getKey());
+                // for single table mv, nested mv's statistics will be more accurate
+                if (baseStatistics.isPresent()
+                        && mv.getBaseTableInfos() != null
+                        && mv.getBaseTableInfos().size() == 1
+                        && !builder.getTableRowCountMayInaccurate()) {
+                    // update based mv output row count to nested mv
+                    Statistics.Builder baseBuilder = Statistics.buildFrom(baseStatistics.get());
+                    baseBuilder.setOutputRowCount(builder.getOutputRowCount());
+                    group.setMvStatistics(entry.getKey(), baseBuilder.build());
+                }
+            }
+        }
+    }
+
+    private boolean isRewrittenMvGE(Operator node, Table table, ExpressionContext context) {
+        return table.isMaterializedView()
+                && node instanceof LogicalOlapScanOperator
+                && context.getGroupExpression() != null
+                && node.getOpType() != context.getGroupExpression().getGroup().getFirstLogicalExpression().getOp().getOpType();
```

Comment:
This may a bit hard code. 

Can we use GrouppExpression#hasAppliedMVRules instead?